### PR TITLE
[DIT-4690] Add Rich Text to Components

### PR DESCRIPTION
**Pull request by asnewman**
## Overview
This PR adds the ability to display rich text on components through the `Ditto` component and `DittoComponent` component by adding the property `richText`.

## Context
Relates to https://github.com/dittowords/ditto-react/issues/34

## Screenshots
image

## Test Plan
The example project has some data setup to display rich text
- [x] Verify that the existing setup shows rich text for `<Ditto componentId="shopping-cart" variables={{ itemCount: 5 }} richText />` (non-plural) and `<Ditto componentId="shopping-cart" count={5} variables={{ itemCount: 5 }} richText />` (plural)
- [x] Update `example/src/ditto/index.js` to reference `ditto-component-library__base__flat.json` instead of `ditto-component-library__base.json` and verify rich text appears for the flat format (components without `richText` enabled will look funky, that is expected)

### CODE REVIEW

File: src/components/DittoComponent.tsx [hunk 1]
```diff
@@ -3,17 +3,27 @@ import { useDittoComponent } from "../hooks/useDittoComponent";
...
   if (typeof text === "string" && typeof children === "function") {
+    if (richText) {
+      return <>{children(text)}</>;
+    }
+
     return <>{children(text)}</>;
   }
 
+  if (richText) {
+    console.log(text)
```
Comment (asnewman) [inline src/components/DittoComponent.tsx:24]:
```suggestion
```

File: src/components/DittoComponent.tsx [hunk 2]
```diff
@@ -3,17 +3,26 @@ import { useDittoComponent } from "../hooks/useDittoComponent";
...
   if (typeof text === "string" && typeof children === "function") {
+    if (richText) {
+      return <>{children(text)}</>;
+    }
+
     return <>{children(text)}</>;
   }
 
+  if (richText) {
+    return <span dangerouslySetInnerHTML={{__html: text || "" }} />;
```
Comment (asnewman) [inline src/components/DittoComponent.tsx:24]: I believe `dangerouslySetInnerHTML` is inevitable here, but if you have any alternative suggestions I'd be glad to hear it.

Comment (jaerod95) [reply src/components/DittoComponent.tsx:20]: hmm....i wonder if this will be a problem with some of our clients.

Comment (jaerod95) [reply src/components/DittoComponent.tsx:20]: maybe we should sanitize the html to be safe? `https://www.npmjs.com/package/dompurify`

**Review by jaerod95 — APPROVED** [review]
Reviewed the new changes and everything seems to be working as expected now! Approving

**Pull request merged**